### PR TITLE
Add wgpu-native

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@ngam which version of OSX do i need for metal to be found?

File: recipes/webgpu-headers/bld.bat [hunk 1]
```diff
@@ -0,0 +1,9 @@
+set CARGO_PKG_VERSION=%PKG_VERSION%
```
Comment (hmaarrfk) [inline recipes/webgpu-headers/bld.bat:1]:
```suggestion
setx CARGO_PKG_VERSION %PKG_VERSION%
```

File: recipes/wgpu-native/bld.bat [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
+if %ERRORLEVEL% neq 0 exit 1
+
+cargo build --release --all-features
+if %ERRORLEVEL% neq 0 exit 1
+
+copy ffi\\wgpu.h %PREFIX%\\Library\\include\\wgpu.h
+if %ERRORLEVEL% neq 0 exit 1
+
+dir target\\release
```
Comment (hmaarrfk) [inline recipes/wgpu-native/bld.bat:14]:
```suggestion
```

File: recipes/wgpu-native/bld.bat [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+copy target\\release\\wgpu_native.dll %PREFIX%\\Library\\bin\\wgpu_native.dll
```
Comment (hmaarrfk) [inline recipes/wgpu-native/bld.bat:15]:
```suggestion
copy target\\release\\wgpu_native.dll %PREFIX%\\Library\\bin\\wgpu_native.dll
if %ERRORLEVEL% neq 0 exit 1
```

File: recipes/wgpu-native/bld.bat [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+copy target\\release\\wgpu_native.lib %PREFIX%\\Library\\lib\\wgpu_native.lib
```
Comment (hmaarrfk) [inline recipes/wgpu-native/bld.bat:16]:
```suggestion
copy target\\release\\wgpu_native.lib %PREFIX%\\Library\\lib\\wgpu_native.lib
if %ERRORLEVEL% neq 0 exit 1
```

**Comment by hmaarrfk** [comment]
@conda-forge/help-rust @conda-forge/help-c-cpp this is ready for review

**Comment by hmaarrfk** [comment]
cc: @conda-forge/rust just wanted to give a ping since apparently help-rust doesn't exist. Thanks for helping review this package.

**Review by xhochy — APPROVED** [review]
Looks good from the rust perspective but I cannot judge the OSX requirements here.

**Comment by hmaarrfk** [comment]
i'm not sure if this compiles anymore, checking again, thanks for chiming in.

**Pull request merged**